Userland/Services/WebContent/RenderLoopExecutor.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2024, Aliaksandr Kalenik <kalenik.aliaksandr@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/EventLoop.h>
+#include <LibGfx/ShareableBitmap.h>
+#include <LibWeb/Painting/PaintingCommandExecutorCPU.h>
+#include <WebContent/RenderLoopExecutor.h>
+
+#ifdef HAS_ACCELERATED_GRAPHICS
+#    include <LibWeb/Painting/PaintingCommandExecutorGPU.h>
+#endif
+
+namespace WebContent {
+
+RenderLoopExecutor::RenderLoopExecutor(bool use_gpu_painter)
+    : m_main_thread_event_loop(Core::EventLoop::current())
+    , m_use_gpu_painter(use_gpu_painter)
+{
+}
+
+RenderLoopExecutor::~RenderLoopExecutor()
+{
+    m_exit = true;
+    m_paint_task_ready_wake_condition.signal();
+}
+
+void RenderLoopExecutor::paint(NonnullOwnPtr<Web::Painting::RecordingPainter> recording_painter, Function<void(i32)> on_completion)
```

Comment:
It seems clunky to pass the whole `RecordingPainter` over to the other thread.
We don't want to do any more recording! Could we pass just the command list instead?

Reply:
yeah, it is indeed clunky to save the whole RecordingPainter in painting task only because it has `execute()` method. it seems like I need to separate "recorded list" into a separate class.